ParkingLot/SmartParkingBoy.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ParkingLotSystem
+{
+    public class SmartParkingBoy : ParkingBoy
+    {
+        public SmartParkingBoy()
+        {
+        }
+
+        public SmartParkingBoy(string name, List<ParkingLot> parkingLotList) : base(name, parkingLotList)
+        {
+        }
+
+        public new string Park(string numberPlate, List<ParkingLot> parkingLots)
```

Comment:
you don't need this method if it's same with boy's Park method, because you have override the ChoseParkingLot.